src/Control/Middleware/RewriteHashLinksMiddleware.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace SilverStripe\Control\Middleware;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\Middleware\HTTPMiddleware;
+use SilverStripe\Core\Convert;
+use SilverStripe\Core\Config\Configurable;
+
+class RewriteHashLinksMiddleware implements HTTPMiddleware
+{
+    use Configurable;
+
+    /**
+     * Mime types to be handled by this middleware
+     * @var string
+     * @config SilverStripe\Control\Middleware\RewriteHashLinksMiddleware.handled_mime_types
+     */
+    private static $handled_mime_types = [
+        'text/html',
+        'application/xhtml+xml',
+    ];
+
+
+    /**
+     * Rewrites hash links in html responses
+     * @param HTTPRequest $request
+     * @param callable $delegate
+     * @return \SilverStripe\Control\HTTPResponse
+     */
+    public function process(HTTPRequest $request, callable $delegate)
+    {
+        /** @var \SilverStripe\Control\HTTPResponse $response **/
+        $response = $delegate($request);
+
+
+        $contentType = explode(';', $response->getHeader('content-type'));
+        $contentType = strtolower(trim(array_shift($contentType)));
+        if (in_array($contentType, $this->config()->handled_mime_types)) {
```

Comment:
```suggestion
        if (!in_array($contentType, $this->config()->handled_mime_types)) {
          return;
        }  
```
And unindent the lines below